test/models/drive_test.dart

```diff
@@ -0,0 +1,1003 @@
+                  driverId: 2,
+                  startTime: DateTime.now().add(const Duration(hours: 16)),
+                  endTime: DateTime.now().add(const Duration(hours: 20)),
+                )
+                .toJsonForApi(),
+            DriveFactory()
+                .generateFake(
+                  driverId: 2,
+                  startTime: DateTime.now().add(const Duration(hours: 12)),
+                  endTime: DateTime.now().add(const Duration(hours: 15)),
+                  cancelled: true,
+                )
+                .toJsonForApi(),
+          ]));
+      expect(
+          await Drive.userHasDriveAtTimeRange(
+              DateTimeRange(
+                start: DateTime.now().add(const Duration(hours: 12)),
+                end: DateTime.now().add(const Duration(hours: 15)),
+              ),
+              2),
+          false);
+    });
+  });
+  group('Drive.getMaxUsedSeats', () {
+    test('returns null if rides are null', () async {
+      final Drive drive = DriveFactory().generateFake(createDependencies: false);
+      expect(drive.getMaxUsedSeats(), null);
+    });
+    test('returns 0 if there is no approved ride', () async {
+      final Drive drive = DriveFactory().generateFake(seats: 3, rides: <Ride>[
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.pending,
+            seats: 1),
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.pending,
+            seats: 3),
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.cancelledByRider,
+            seats: 2),
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.rejected,
+            seats: 2),
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.withdrawnByRider,
+            seats: 1),
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.cancelledByDriver,
+            seats: 2),
+      ]);
+      expect(drive.getMaxUsedSeats(), 0);
+    });
+    test('returns 1 if rides has only 1 approved ride with 1 seat', () async {
+      final Drive drive = DriveFactory().generateFake(seats: 3, rides: <Ride>[
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.approved,
+            seats: 1),
+        RideFactory().generateFake(
```

Comment:
IMO, all of these further rides are unnecessary. We have established in the previous test that every ride with a status != approved will be counted as having 0 seats, so this is clutter